app/vmstorage/servers/vmselect.go

```diff
@@ -35,6 +38,11 @@ var (
 		"This may be useful when multiple data sources with distinct retentions are hidden behind query-tee")
 )
 
+var (
+	defaultMaxUniqueTimeseries int
```

Comment:
I think it's better to rename `defaultMaxUniqueTimeseries` to `maxUniqueTimeseriesValue`, since it's variable and not a constant value.

 Also, it's better to assign name of the variable + once to the mutex protecting it's initialisation:
`once` -> `maxUniqueTimeseriesValueOnce`. 
 It should make code easy to understand for readers.